packages/indy-sdk-to-askar-migration/src/IndySdkToAskarMigrationUpdater.ts

```diff
@@ -0,0 +1,198 @@
+        break
+      }
+      if (ms.length > 1) throw new Error('Multiple Master Secrets found!')
+
+      const row = ms[0]
+      await txn.remove({ category, name: row.name })
+      const value = (typeof row.value === 'string' ? JSON.parse(row.value) : row.value).value
+
+      await txn.insert({ category: 'master_secret', name: row.name, value })
+      updateCount++
+
+      await txn.commit()
+    }
+
+    console.log(`Updated ${updateCount} instances inside ${category}`)
+  }
+
+  private async updateDids() {
+    const category = 'Indy::Did'
+    const categoryMeta = 'Indy::DidMetadata'
+
+    console.log(`Updating ${category}`)
+
+    let updateCount = 0
+    const session = this.store.transaction()
+    for (;;) {
+      const txn = await session.open()
+      const dids = await txn.fetchAll({ category, limit: 50 })
+      if (!dids || dids.length === 0) {
+        await txn.close()
+        break
+      }
+
+      for (const row of dids) {
+        await txn.remove({ category, name: row.name })
+        const info = typeof row.value === 'string' ? JSON.parse(row.value) : row.value
+        console.log(info)
+        let meta
+        try {
+          meta = await txn.fetch({
+            category: categoryMeta,
+            name: row.name,
+          })
+        } catch {}
+        let didMeta
+        if (meta) {
+          await txn.remove({ category: categoryMeta, name: meta.name })
+          didMeta = (typeof meta.value === 'string' ? JSON.parse(meta.value) : meta.value).value
+        }
+        await txn.insert({
+          category: 'did',
+          name: row.name,
+          value: {
+            verkey: info.verkey,
+            did: info.did,
+            metadata: didMeta,
+          },
+          tags: { verkey: info.verkey },
+        })
+        updateCount++
+      }
+      await txn.commit()
+    }
+
+    console.log(`Updated ${updateCount} instances inside ${category}`)
+  }
+
+  private async updateCredentials() {
+    const category = 'Indy::Credential'
+
+    console.log(`Updating ${category}`)
+
+    let updateCount = 0
+    const session = this.store.transaction()
+    for (;;) {
+      const txn = await session.open()
+      const credentials = await txn.fetchAll({ category, limit: 50 })
+      if (!credentials || credentials.length === 0) {
+        await txn.close()
+        break
+      }
+
+      for (const row of credentials) {
+        await txn.remove({ category, name: row.name })
+        const credentialData = typeof row.value === 'string' ? JSON.parse(row.value) : row.value
+        const tags = this.credentialTags(credentialData)
+        await txn.insert({
+          category: 'credential',
+          name: row.name,
+          value: row.value,
+          tags,
+        })
+        updateCount++
+      }
+      await txn.commit()
+    }
+
+    console.log(`Updated ${updateCount} instances inside ${category}`)
+  }
+
+  private credentialTags(credentialData: Record<string, unknown>) {
```

Comment:
@TimoGlastra You mentioned to me this would be good to have in another package. Where would that be?

Reply:
We have the `AnonCredsCredentialRecord`. that if you create it with all the required fields, you can call `getTags` and it will automatically have the correct format